code/modules/awaymissions/mission_code/dsbssigma.dm

```diff
@@ -0,0 +1,735 @@
+	/obj/item/clothing/under/chameleon = 15)
+
+/// MOBS ///
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob
+	icon = 'icons/misc/mark.dmi'
+	icon_state = "r_mob"
+	color = "#c80004"
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien
+	name = "random alien spawner"
+	result = list(
+	/mob/living/simple_animal/hostile/alien = 1,
+	/mob/living/simple_animal/hostile/alien/drone = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien_guard
+	name = "random alien guard spawner"
+	color = "#960018"
+	result = list(
+	/mob/living/simple_animal/hostile/alien/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel/stationary = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien_50pc
+	name = "random alien spawner 50pc"
+	color = "#ff7700"
+	result = list(
+	/datum/nothing = 3,
+	/mob/living/simple_animal/hostile/alien = 1,
+	/mob/living/simple_animal/hostile/alien/drone = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien_30pc
+	name = "random alien spawner 30pc"
+	color = "#ffc800"
+	result = list(
+	/datum/nothing = 7,
+	/mob/living/simple_animal/hostile/alien = 1,
+	/mob/living/simple_animal/hostile/alien/drone = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien_stationary
+	name = "random stationary alien spawner"
+	result = list(
+	/mob/living/simple_animal/hostile/alien/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/drone/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel/stationary = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien_stationary_50pc
+	name = "random stationary alien spawner 50pc"
+	color = "#ff7700"
+	result = list(
+	/datum/nothing = 3,
+	/mob/living/simple_animal/hostile/alien/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/drone/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel/stationary = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/alien_stationary_30pc
+	name = "random stationary alien spawner 30pc"
+	color = "#ffc800"
+	result = list(
+	/datum/nothing = 7,
+	/mob/living/simple_animal/hostile/alien/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/drone/stationary = 1,
+	/mob/living/simple_animal/hostile/alien/sentinel/stationary = 1)
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/mob/dead_monkey_50pc
+	name = "dead monkey 50pc"
+	color = "#0e7500"
+	result = list(
+	/datum/nothing = 1,
+	/obj/effect/mob_spawn/human/corpse/away/dsbssigma/monkey = 1)
+
+/// RANDOMIZER ///
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/randomizer
+	name = "spawn randomizer"
+	icon_state = "x"
+	color = "#800080"
+	early_del = FALSE
+	var/r_location = null
+	var/r_object = null
+
+/obj/effect/spawner/random_spawners/away/dsbssigma/randomizer/proc/randomize(r_location, r_object)
```

Comment:
I would remove this proc because it's looping through `GLOB.landmarks_list` more times than is necessary. 

Instead, in `LateInitialize` I would loop through it and make and fill an associative list with the key being the landmark name and the value a list of landmarks with that name (`/text` => `/list/obj/effect/landmark`). Only add those which match any of the relevant landmark names. 
Then to spawn an object you can simply `pick`.

A brief, untested code would be:

```dm
#define DBSSIGMA_LANDMARK_QUEEN "dsbssigma_queen"
// ETC...

var/list/spawn_locations = list()
for(var/thing in GLOB.landmarks_list)
	var/obj/effect/landmark/L = thing
	if(L.name == DBSSIGMA_LANDMARK_QUEEN /* ETC.. */)
		spawn_locations[L.name] += list(get_turf(L))

/*
	From here you have a list of list of turfs (corresponding to the landmark on it), organized per landmark name
	It would look like:
	var/list/spawn_locations = list(
		"dsbssigma_queen" = list(
			/turf (instantiated),
			...
		),
		...
	)
*/

// Spawn stuff using our newly populated list
new /mob/living/simple_animal/hostile/alien/queen/large/dsbssigma(pick(relevant_landmarks[DBSSIGMA_LANDMARK_QUEEN]))

#undef DBSSIGMA_LANDMARK_QUEEN
```